src/main/kotlin/org/rust/lang/core/resolve/ref/RsResolveCache.kt

```diff
@@ -49,26 +55,53 @@ class RsResolveCache(messageBus: MessageBus) {
      * Expected resolve results: [PsiElement], [ResolveResult] or a [List]/[Array] of [ResolveResult]
      */
     @Suppress("UNCHECKED_CAST")
-    fun <K: Any, V> resolveWithCaching(key: K, resolver: (K) -> V): V? {
+    fun <K : PsiElement, V> resolveWithCaching(key: K, resolver: (K) -> V): V? {
         ProgressManager.checkCanceled()
-        return cache[key] as V? ?: run {
+        val map = getCacheFor(key)
+        return map[key] as V? ?: run {
```

Comment:
Isn't it's better to use `as? V` instead of `as V?`? If `map` contains object of type  other than `V`, this will cause `ClassCastException`

Reply:
ClassCastException will be anyway because of type erasure